File: lib/round-play.ts
export type RoundPlayMode = 'full' | 'front9' | 'back9'

export function getRoundPlayMode(courseHoleCount: number, holeNumbers: number[]): RoundPlayMode {
  if (courseHoleCount === 18 && holeNumbers.length === 9) {
    if (holeNumbers[0] === 1 && holeNumbers.at(-1) === 9) {
      return 'front9'
    }

    if (holeNumbers[0] === 10 && holeNumbers.at(-1) === 18) {
      return 'back9'
    }
  }

  return 'full'
}

export function getRoundPlayLabel(courseHoleCount: number, holeNumbers: number[]) {
  const mode = getRoundPlayMode(courseHoleCount, holeNumbers)

  if (mode === 'front9') return 'Front 9'
  if (mode === 'back9') return 'Back 9'
  return courseHoleCount === 9 ? '9 Holes' : 'Full Course'
}

export function getCurrentHolePosition(holeNumbers: number[], currentHole: number) {
  const index = holeNumbers.indexOf(currentHole)
  return index === -1 ? 1 : index + 1
}

File: rounds/round_play.py
def get_current_hole_position(hole_numbers: list[int], current_hole: int) -> int:
    """Return the 1-based position of current_hole within hole_numbers."""
    try:
        return hole_numbers.index(current_hole) + 1
    except ValueError:
        return 1


def get_round_play_label_from_mode(play_mode: str, course_hole_count: int) -> str:
    if play_mode == "front9":
        return "Front 9"
    if play_mode == "back9":
        return "Back 9"
    return "9 Holes" if course_hole_count == 9 else "Full Course"
